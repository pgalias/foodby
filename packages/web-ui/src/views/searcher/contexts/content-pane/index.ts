export {
  ContentPaneProvider,
  useChangeCurrentPaneState,
  useCurrentPaneState,
  States,
} from './provider';
