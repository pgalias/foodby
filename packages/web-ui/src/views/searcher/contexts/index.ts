export * from './content-pane';
export * from './filters';
