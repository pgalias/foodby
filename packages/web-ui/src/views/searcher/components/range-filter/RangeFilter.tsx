import React, { FC, useState } from 'react';
import { Card, Slider } from '@foodby/common-components-ui';
import clsx from 'clsx';
import { Types, useFilterDispatch, useFilters } from '../../contexts';

export const RangeFilter: FC = () => {
  const { radius } = useFilters();
  const dispatch = useFilterDispatch();
  const [value, setValue] = useState(radius / 1000);

  const onChange = (newValue: number) => {
    setValue(newValue);
    dispatch({
      type: Types.SET_RADIUS,
      payload: newValue * 1000, // value have to be passed in meters
    });
  };

  return (
    <Card className={clsx('mx-auto', 'h-48', 'center-y')}>
      <Slider
        min={1}
        max={40}
        onChange={onChange}
        step={0.1}
        label="Range (km)"
        value={value}
      />
    </Card>
  );
};
