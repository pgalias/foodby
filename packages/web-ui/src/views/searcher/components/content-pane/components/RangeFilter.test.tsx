import * as React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { RangeFilter } from './RangeFilter';
import { Types } from '../../../contexts';

const mockDispatch = jest.fn();

jest.mock('../../../contexts', () => ({
  ...(jest.requireActual('../../../contexts') as Record<string, unknown>),
  useFilterDispatch: () => mockDispatch,
  useFilters: jest.fn(() => ({
    radius: 2500,
  })),
}));

describe('RangeFilter', () => {
  test('should render a card and slider components', () => {
    const { asFragment } = render(<RangeFilter />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should dispatch an filter action on slider value change', () => {
    render(<RangeFilter />);
    fireEvent.change(screen.getByRole('slider'), { target: { value: '4.5' } });

    expect(mockDispatch).toHaveBeenCalledWith({
      type: Types.SET_RADIUS,
      payload: 4500,
    });
  });

  test('should create a slider with initial value from current filters store', () => {
    render(<RangeFilter />);

    expect(screen.getByRole('note').textContent).toEqual('2.5');
  });
});
