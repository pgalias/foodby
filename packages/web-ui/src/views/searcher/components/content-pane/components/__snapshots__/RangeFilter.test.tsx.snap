// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeFilter should render a card and slider components 1`] = `
<DocumentFragment>
  <section
    class="fb-card w-11/12 md:w-3/5 lg:w-2/5 p-8 bg-white text-blueGray-600 rounded-lg overflow-hidden mx-auto h-48 center-y"
  >
    <div
      class="fb-slider"
    >
      <label
        class="text-sm text-coolGray-400 mb-4 block"
        for="Slider_1"
      >
        Range (km)
      </label>
      <div
        class="flex justify-between items-center"
      >
        <span
          class="text-sm text-coolGray-500 text-center pr-2"
        >
          1
        </span>
        <input
          class="_Slider_p92bp_1"
          id="Slider_1"
          max="40"
          min="1"
          step="0.1"
          type="range"
          value="2.5"
        />
        <span
          class="text-sm text-coolGray-500 text-center pl-2"
        >
          40
        </span>
      </div>
      <span
        aria-live="polite"
        class="text-sm text-coolGray-500 text-center block pt-1"
        role="note"
      >
        2.5
      </span>
    </div>
  </section>
</DocumentFragment>
`;
